flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/utils/RecordReaderIterator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.utils;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+/** Wrap a {@link RecordReader} as an {@link Iterator}. */
+public class RecordReaderIterator<T> implements Iterator<T> {
```

Comment:
Just a util method like `org.apache.flink.connector.file.src.util.Utils.forEachRemaining`

Reply:
I'm actually planning to replace these `RecordReader` thingy with `Iterator` in the future so I'll use `Iterator` as much as I can. Also if we want to implement file rolling we cannot directly use `forEachRemaining` method.

Reply:
> replace these RecordReader thingy with Iterator in the future

How? You mean `Reader` and `Iterator`? Anyway we need batch iterface.

> Also if we want to implement file rolling we cannot directly use forEachRemaining method

Are they related? One is writing and one is reading.

Reply:
Currently this iterator is used for writing. In the future I only want `SstFile` to deal with blocks. `SortMergeReader` and the components above should only see `Iterator`.